File: src/main/scala/org/dama/datasynth/utils/Dictionary.scala
package org.dama.datasynth.utils

import scala.collection.mutable

/**
  * Created by aprat on 21/12/16.
  */
class Dictionary {
  var hashMap = new mutable.HashMap[String,Int]
  var nextIndex = 0

  def add( value : String ): Unit = {
    if(!hashMap.contains(value)) {
      hashMap.put(value, nextIndex)
      nextIndex += 1
    }
  }

  def get( value : String ) : Int = {
    hashMap.get(value).get
  }

  def apply(value : String) : Int = get(value)

}

File: src/main/java/org/dama/datasynth/runtime/spark/operators/matching/Dictionary.java
package org.dama.datasynth.runtime.spark.operators.matching;

import java.util.HashMap;

/**
 * Created by aprat on 6/03/17.
 */
public class Dictionary<T extends Comparable<T>, S extends Comparable<S>> {

    private HashMap<T,S> data = new HashMap<>();

    public Dictionary(Table<T,S> table) {
        for( Tuple<T,S> entry : table) {
            data.put(entry.getX(),entry.getY());
        }
    }

    public S get(T key) {
        return data.get(key);
    }
}
